lib/cylc/task_proxy.py

```diff
@@ -1508,9 +1508,13 @@ def get_state_summary(self):
         """Return a dict containing the state summary of this task proxy."""
         self.summary['state'] = self.state.status
         self.summary['spawned'] = str(self.has_spawned)
-        self.summary['mean_elapsed_time'] = (
-            float(sum(self.tdef.elapsed_times)) /
-            max(len(self.tdef.elapsed_times), 1))
+        count = len(self.tdef.elapsed_times)
+        if count == 0:
+            self.summary['mean_elapsed_time'] = \
+                max(self.summary['execution_time_limit'], 1)
+        else:
+            self.summary['mean_elapsed_time'] = (
+                float(sum(self.tdef.elapsed_times)) / count)
```

Comment:
I like the general idea but I'm not sure I understand the 0 vs 1 value - `self.summary['execution_time_limit']` is a duration like `PT10M` here isn't it, so the `max` computation with 0 or 1 doesn't make sense?  And anyway, wouldn't it be best to show no progress bar in the first run of a task if no execution time limit is set?